DQMServices/Core/interface/MonitorElement.h

```diff
@@ -69,176 +96,197 @@ namespace dqm::impl {
+    /// true if ME was updated in last monitoring cycle
+    bool wasUpdated() const { return data_.flags & DQMNet::DQM_PROP_NEW; }
+    void packScalarData(std::string &into, const char *prefix) const;
+    void packQualityData(std::string &into) const;
+    DQMNet::CoreObject data_;  //< Core object information.
+
+  public:
+    MonitorElement &operator=(const MonitorElement &) = delete;
+    MonitorElement &operator=(MonitorElement &&) = delete;
+    virtual ~MonitorElement();
+
+  public:
+    // good to be used in subsystem code
 
     /// Get the type of the monitor element.
     Kind kind() const { return Kind(data_.flags & DQMNet::DQM_PROP_TYPE_MASK); }
 
-    /// Get the object flags.
-    uint32_t flags() const { return data_.flags; }
-
     /// get name of ME
-    const std::string &getName() const { return data_.objname; }
+    const std::string &getName() const { return this->data_.objname; }
 
     /// get pathname of parent folder
-    const std::string &getPathname() const { return *data_.dirname; }
+    const std::string &getPathname() const { return this->data_.dirname; }
 
     /// get full name of ME including Pathname
-    const std::string getFullname() const {
-      std::string path;
-      path.reserve(data_.dirname->size() + data_.objname.size() + 2);
-      path += *data_.dirname;
-      if (!data_.dirname->empty())
-        path += '/';
-      path += data_.objname;
-      return path;
-    }
-
-    /// true if ME was updated in last monitoring cycle
-    bool wasUpdated() const { return data_.flags & DQMNet::DQM_PROP_NEW; }
+    std::string getFullname() const { return access().key.path_.getFullname(); }
 
-    /// Mark the object updated.
-    void update() { data_.flags |= DQMNet::DQM_PROP_NEW; }
+    edm::LuminosityBlockID getRunLumi() { return access().key.id_; }
 
-    /// specify whether ME should be reset at end of monitoring cycle (default:false);
-    /// (typically called by Sources that control the original ME)
-    void setResetMe(bool /* flag */) { data_.flags |= DQMNet::DQM_PROP_RESET; }
+    MonitorElementData::Scope getScope() { return access().key.scope_; }
 
     /// true if ME is meant to be stored for each luminosity section
-    bool getLumiFlag() const { return data_.flags & DQMNet::DQM_PROP_LUMI; }
-
-    /// this ME is meant to be stored for each luminosity section
-    void setLumiFlag() { data_.flags |= DQMNet::DQM_PROP_LUMI; }
+    bool getLumiFlag() const { return access().key.scope_ == MonitorElementData::Scope::LUMI; }
 
     /// this ME is meant to be an efficiency plot that must not be
     /// normalized when drawn in the DQM GUI.
-    void setEfficiencyFlag() { data_.flags |= DQMNet::DQM_PROP_EFFICIENCY_PLOT; }
-    bool getEfficiencyFlag() { return data_.flags & DQMNet::DQM_PROP_EFFICIENCY_PLOT; }
+    void setEfficiencyFlag() {
+      auto access = this->accessMut();
+      if (access.value.object_)
+        access.value.object_->SetBit(TH1::kIsAverage);
```

Comment:
@mtosi Looks like Github screwed up that link before. This is the relevant line.

Reply:
thanks, but I'm sorry, I'm still confused
the single histogram used by the stacked plot is an efficiency, right, 
but it comes from
https://github.com/cms-sw/cmssw/blob/CMSSW_11_1_0_pre3/Validation/RecoTrack/python/PostProcessorTracker_cfi.py#L21
which is making use of the `DQMGenericClient`

what am I missing ?
thanks

Reply:
forget, I missed your message above
and indeed, I'm trying to rollback this bit, but I'm not able to :(
how should this be done ?
thanks

Reply:
`th1.SetBit(ROOT.kIsAverage, 0)` or `th1.ResetBit(ROOT.kIsAverage)` would be my guess in Python. If you have trouble finding the constant, you can just use the value directly (`1 << 18`).